msrest/paging.py

```diff
@@ -73,22 +70,38 @@ def raw(self):
         return raw
 
     def get(self, url):
-        """Get arbitrary page.
+        """Get an arbitrary page.
+
+        This resets the iterator and then fully consumes it to return the
+        specified page.
 
         :param str url: URL to arbitrary page results.
         """
+        self.reset()
         self.next_link = url
-        return self.next()
+        return list(self)
```

Comment:
This will not work if there is a page3. Current code will return page 2 + page 3. Yes, I miss this unittest... :(